crates/indexer/src/runtime.rs

```diff
@@ -0,0 +1,208 @@
+        run_forever(indexer_config, context).await;
+    });
+
+    Some(Ok(runtime))
+}
+
+pub async fn run_forever(config: IndexerConfig, context: Arc<Context>) {
+    let processor_name = config.processor.clone();
+
+    info!(processor_name = processor_name, "Starting indexer...");
+
+    let conn_pool = new_db_pool(&config.postgres_uri).expect("Failed to create connection pool");
+    info!(
+        processor_name = processor_name,
+        "Created the connection pool... "
+    );
+
+    info!(processor_name = processor_name, "Instantiating tailer... ");
+
+    let processor_enum = Processor::from_string(&processor_name);
+    let processor: Arc<dyn TransactionProcessor> = match processor_enum {
+        Processor::DefaultProcessor => {
+            Arc::new(DefaultTransactionProcessor::new(conn_pool.clone()))
+        }
+        Processor::TokenProcessor => Arc::new(TokenTransactionProcessor::new(
+            conn_pool.clone(),
+            config.index_token_uri_data,
+        )),
+    };
+
+    let options = TransactionFetcherOptions::new(
+        None,
+        None,
+        Some(config.batch_size),
+        None,
+        config.fetch_tasks as usize,
+    );
+
+    let tailer = Tailer::new(context, conn_pool.clone(), processor, options)
+        .expect("Failed to instantiate tailer");
+
+    if !config.skip_migrations {
+        info!(processor_name = processor_name, "Running migrations...");
+        tailer.run_migrations();
+    }
+
+    let start_version = match config.starting_version {
+        None => tailer
+            .get_start_version(&processor_name)
+            .unwrap_or_else(|| {
+                info!(
+                    processor_name = processor_name,
+                    "Could not fetch version from db so starting from version 0"
+                );
+                0
+            }),
+        Some(version) => version,
+    };
+
+    info!(
+        processor_name = processor_name,
+        start_version = start_version,
+        "Setting starting version..."
+    );
+    tailer.set_fetcher_version(start_version).await;
+
+    info!(processor_name = processor_name, "Starting fetcher...");
+    tailer.transaction_fetcher.lock().await.start().await;
+
+    let start = chrono::Utc::now().naive_utc();
+
+    info!(
+        processor_name = processor_name,
+        start_version = start_version,
+        "Indexing loop started!"
+    );
+
+    let mut versions_processed: u64 = 0;
+    let mut total_processed: u64 = 0;
+    let mut base: u64 = 0;
+    let mut version_to_check_chain_id: u64 = 0;
+
+    // Check once here to avoid a boolean check every iteration
+    if config.check_chain_id {
+        tailer
+            .check_or_update_chain_id()
+            .await
+            .expect("Failed to get chain ID");
+        version_to_check_chain_id = versions_processed + 100_000;
+    }
+
+    let (tx, mut receiver) = tokio::sync::mpsc::channel(100);
```

Comment:
should this be hardcoded?

Reply:
this one matters the least; it'll never be hit basically